File: codejam_and_others/hoppity/hoppity.py
# hoppity hop
# facebook puzzle #1
import sys

def Hop(file_name):
	with open(file_name) as f:
		# since the file have just one number
		# we read directly
		num = int(f.read())
			
		for i in range(1, num+1):
			if i % 3 == 0 and i % 5 == 0:
				print 'Hop\n',
			elif i % 5 == 0: 
				print 'HopHop\n',
			elif i % 3 == 0 :
				print 'Hoppity\n', 
def main():
	file_name=''
	if len(sys.argv) < 2:
		print 'ops, must pass a file name'
		sys.exit()
	else:
		Hop(sys.argv[1])

if __name__=='__main__':
	main()
	
#~ if len(sys.argv) < 2: sys.exit('ops! must pass a file name')
#~ with open(sys.argv[1]) as f: 
    #~ for i in range(1,int(f.read())+1):
        #~ if i % 3 == 0 and i % 5 == 0: print 'Hop\n',
        #~ elif i % 5 == 0: print 'HopHop\n',
        #~ elif i % 3 == 0 : print 'Hoppity\n',

File: codejam_and_others/hoppity/hoppity.c
//      Hoppity facebook puzzle in C


#include <stdio.h>

int main(int argc, char **argv)
{
	if (argc < 2){
        printf("Ops! need a file name argument \n\n");
        return -1;
    }
    
    FILE *f;
    char line[80];
    int num;
    
    f = fopen(argv[1], "r");
    fgets(line, 80, f);
    printf("\n%s \n\n", line);
    num =  atoi(line);
    
    int i ;
    for (i = 1; i<=num ; i++){
        if (i % 3 == 0 && i % 5 == 0)
            printf("Hop\n");
        else if (i % 5 == 0)
            printf("HopHop\n");
        else if (i % 3 == 0)
            printf("Hoppity\n"); 
    }
    
    
	return 0;
}

